Basic/question_3.java

```diff
@@ -0,0 +1,190 @@
+// Four pillars of oop are :
+// 1. Abstraction: The process of hiding the unnecessary details of an object and only exposing the essential information. It can be acheived using Abstract classes or Interface.
+//abstract class(0-100% Abstraction means abtract and concrete both methods can)  
+abstract class Demo  
+{  
+//abstract method  
+abstract void display();  
+}  
+//extends the abstract class(Must , otherwise the abtract class object cannot be instantiate)
+public class MainClass extends Demo  
+{  
+
+void display()  
+{  
+System.out.println("Abstract method called.");  
+}  
+public static void main(String[] args)  
+{    
+MainClass obj = new MainClass ();  
+//invoking abstract method  
+obj.display();  
+}  
+} 
+
+// Output: Abstract method called
+//        
+
+
+
+//Interfece(100% Abstraction )
+
+Interface Demo  
+{  
+
+ void greeting();
+ void display();  
+}  
+//Implements  the Interface(Must implement all methods of Interface)
+public class MainClass implements Demo  
+{  
+
+void greeting()
+{
+System.out.println("Hey this is the Implementation of Interface class Method.");  
+
+}
+void display()  
+{  
+System.out.println("Interface class method called.");  
+}  
+public static void main(String[] args)  
+{    
+MainClass obj = new MainClass ();  
+//invoking abstract method  
+obj.greeting();
+obj.display();  
+}  
+} 
+
+
+
+
+// Output: Hey this is the Implementation of Interface class Method.
+//         Interface class method called.
+
+// 2. Encapsulation:The process of binding fields (object state) and methods (behavior) together as a single unit.In Java it is done using classes.
```

Comment:
Make your comments consistent, like on line **63** if you have decided to cut a long line into two lines. Do the same thing here too